### PR TITLE
add the ability to sort multiple columns insensitively

### PR DESCRIPTION
**Pull request by fdncred**
# Description

this pr adds the ability to sort multiple columns insensitively as in `ls | sort-by name type -i`. it's a draft because i may have done something unholy by hacking together an `impl Ord for Value` and `impl Eq for Value`. So, i'm expecting what i did to be disallowed. :)
  
# Tests

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

**Comment by stormasm** [comment]
@fdncred ok I reviewed the code and this is great !  

I don't know if its correct, we will let @jntrnr see what he thinks... 

but this is a BIG step forward from what I have done as it implements Ord which needed to be done, and I wasn't sure how to do it...

also, this will enable JT's desire to get the Value::List sorted as well...
which should then close out  #4336 and accomplish the List sorting...

One question I have is my arbitrary sort order will stay in place for lists

/*
Arbitrary Order of Values:
    Floats
    Ints
    Strings
    Bools
    Lists
*/

But I guess this is not needed for table columns as all of the types
in the tables will be the same / and / or the Ord will take care of the
sorting between different Value types ?

**Comment by fdncred** [comment]
@stormasm i'm just guessing too - i'm not sure if what I've done is right and land-able or if it needs tweaks, or just needs to be trashed. Really what i tried to do is make sort look more like nushell 0.44's sort.

File: crates/nu-protocol/src/value/mod.rs [hunk 1]
```diff
@@ -953,6 +953,60 @@ impl PartialEq for Value {
...
+                    ..
+                },
+                Value::Record {
+                    vals: rhs,
+                    cols: rhs_headers,
+                    ..
+                },
+            ) if lhs_headers == rhs_headers && lhs == rhs => Ordering::Equal,
+            (Value::Binary { val: lhs, .. }, Value::Binary { val: rhs, .. }) => lhs.cmp(rhs),
+            // (Value::CustomValue { val: lhs, .. }, rhs) => lhs.cmp(rhs),
```
Comment (jntrnr) [inline crates/nu-protocol/src/value/mod.rs:1001]: Can you make a note in an issue that CustomValue will need Ord in the future?

**Comment by fdncred** [comment]
close in favor of #4498 and a couple others

**Pull request closed without merging**